include/zephyr/net/coap.h

```diff
@@ -1164,6 +1164,32 @@ int coap_resource_notify(struct coap_resource *resource);
  */
 bool coap_request_is_observe(const struct coap_packet *request);
 
+/**
+ * @brief CoAP transmission configuration
+ */
+struct coap_config_t {
```

Comment:
We don't have `_t` appended to struct types. Also `coap_config(_t)` seems too broad for what you're trying to achieve here. These are settings specific to retransmissions, so `struct coap_pending` options.